server/src/main/java/org/opensearch/common/util/concurrent/OpenSearchExecutors.java

```diff
@@ -137,6 +137,19 @@ public static OpenSearchThreadPoolExecutor newFixed(String name, int size, int q
             queue, threadFactory, new OpenSearchAbortPolicy(), contextHolder);
     }
 
+    public static OpenSearchThreadPoolExecutor newResizable(String name, int size, int queueCapacity,
+                                                    ThreadFactory threadFactory, ThreadContext contextHolder) {
+        BlockingQueue<Runnable> queue;
+        if (queueCapacity < 0) {
```

Comment:
does it make any sense to be ok with a queue size of 0?

Reply:
yes, Queue size can be set to -1 to make it unbounded. This is to handle the case where queue size is -1 in opensearch.yml

Reply:
I am asking specifically about 0 though. I get the need for a -1 (although a bit confusing) as a flag for unbounded. I am asking to understand more than any form of critique of course:)